geotrek/trekking/parsers.py

```diff
@@ -1014,13 +1014,24 @@ def _convert_to_geos(geom):
     def _maybe_get_linestring_from_layer(layer):
         if layer.num_feat == 0:
             return None
-        first_feature = layer[0]
-        geos = ApidaeTrekParser._convert_to_geos(first_feature.geom)
-        if geos.geom_type == 'MultiLineString':
-            geos = geos.merged
+        geoms = []
+        for feat in layer:
+            if feat.geom.num_coords == 0:
+                continue
+            geos = ApidaeTrekParser._convert_to_geos(feat.geom)
             if geos.geom_type == 'MultiLineString':
-                raise RowImportError(_("The geometry cannot be converted to a single continuous LineString feature"))
-        return geos
+                geos = geos.merged
+                if geos.geom_type == 'MultiLineString':
+                    raise RowImportError(_("Feature geometry cannot be converted to a single continuous LineString feature"))
+            geoms.append(geos)
+
+        full_geom = MultiLineString(geoms)
+        full_geom.srid = geoms[0].srid
+        full_geom = full_geom.merged
```

Comment:
Write an comment to explain merged operation

Reply:
done